src/Bicep.Core/Syntax/SyntaxTreeFactory.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Azure.Deployments.Core.Definitions.Schema;
+using Azure.Deployments.Core.Json;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Extensions;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Parsing;
+using Bicep.Core.Syntax.Converters;
+using Bicep.Core.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Core.Syntax
+{
+    public static class SyntaxTreeFactory
+    {
+        public static SyntaxTree CreateSyntaxTree(Uri fileUri, string fileContents, bool isEntryFile) => isEntryFile
+            ? CreateBicepSyntaxTree(fileUri, fileContents)
+            : CreateSyntaxTree(fileUri, fileContents);
+
+        public static SyntaxTree CreateSyntaxTree(Uri fileUri, string fileContents) =>
+            PathHelper.HasBicepExtension(fileUri) || !PathHelper.HasAnyExtension(fileUri)
```

Comment:
>HasAnyExtension [](http://example.com/codeflow?start=65&length=15)

Maybe I missed it elsewhere, but we are rather strict in what we allow as a module path. I think we should restrict the extensions we use to reference JSON templates. (We can always add more later, but once we open it up fully, we can never use any other extension in the path.)

Reply:
Good pt. I'll reverse the logic so that it will only create JSON modules if the extension is .{json,jsonc,arm}.